### PR TITLE
Fix Track X3D export.

### LINKED ISSUE #305: Crash when Exporting Robot with Track to X3D
**Describe the Bug**
If Webots is started with the `--enable-x3d-meta-file-export` option it will crash when exporting to X3D a robot with track nodes.

### PR DESCRIPTION
**Pull request by DavidMansolino**
Fix #305

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
:eyes:

**Comment by stefaniapedrazzi** [comment]
Oops.. there was no thumb up..

**Comment by DavidMansolino** [comment]
Indeed, it's not ready (sorry I forget to put 'draft' PR and once it is created you can't change anymore :-/ )

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
:+1:

**Comment by DavidMansolino** [comment]
(result visible here: https://www.cyberbotics.com/doc/guide/telemax?version=feature-telemax-robot)

**Comment by stefaniapedrazzi** [comment]
:eyes:

**Comment by DavidMansolino** [comment]
> Maybe it would be better to list the Track motors in the LinearMotor section with all the other motors given that it is a normal LinearMotor node.

This is what I did initially but the problem is that any node in this section will have the slider to move it, and since our viewer doesn't support moving track node it looks a bit strange to have this useless slider. I will improve it to not show the slider in this case (even so the goal of this PR was initialyl to hotfix the crash and not extend our viewer).

**Comment by stefaniapedrazzi** [comment]
Alternatively if it doesn't look consistent to not have sliders you could maybe just change the section name from "Track" to "Track motors".

**Comment by DavidMansolino** [comment]
> Alternatively if it doesn't look consistent to not have sliders you could maybe just change the section name from "Track" to "Track motors".

That's the first things I tried but our viewer is probably checking for the 'motor' string as it was displaying sliders for this new category too :rofl:

**Comment by DavidMansolino** [comment]
Now they are in the LinearMotor section but without the slider:
Capture du 2019-03-26 16-25-44

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
:+1:

**Comment by stefaniapedrazzi** [comment]
:eyes:

File: src/webots/nodes/utils/WbWorld.cpp [hunk 1]
```diff
@@ -445,6 +445,8 @@ void WbWorld::createX3DMetaFile(const QString &filename) const {
             WbVector3 v = m.translation();
             if (!v.almostEquals(WbVector3()))
               deviceObject.insert("transformOffset", v.toString(WbPrecision::FLOAT_MAX));
+            if (motor)
```
Comment (stefaniapedrazzi) [inline src/webots/nodes/utils/WbWorld.cpp]: It doesn't seem correct to assume that if a device has a motor but it is not a joint, then it is a `Track` device.
A `Propeller` has a motor too, so with the current code the `Propeller` motor is tagged with the `track` flag.
I think that here an additional type check is needed to set the track flag correctly just for Track motors.

Comment (DavidMansolino) [reply src/webots/nodes/utils/WbWorld.cpp]: Very good point, I did forget the propeller case.

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
:+1:

**Pull request merged**